### PR TITLE
cWorld: Move most work from Start into the constructor

### LINKED ISSUE #4042: If plugin manipulates world on startup server crashes
Commit id:        e883aa828c304600fcd707b50df9515011311db8

### Expected behavior
Server should not crash.

### Actual behavior
Server crashes.

### Steps to reproduce the behavior
Load this plugin code:
```
function Initialize(a_Plugin)
	a_Plugin:SetName("Crash")
	cRoot:Get():GetDefaultWorld():SetBlock(0, 100, 0, 1, 0)
	return true
end
```

Ref, forum post:
https://forum.cuberite.org/thread-2937.html

### PR DESCRIPTION
**Pull request by peterbell10**
Fixes #4042. Tested with the script from the issue.

`Start` now does nothing other than starting the world's threads.  This means everything is initialised properly before the plugins are loaded.  Also, as the threads are still not started in the constructor, you keep the guarantee that chunks can't be generated before hooks are registered.

### CODE REVIEW

File: src/World.cpp [hunk 1]
```diff
@@ -146,7 +150,7 @@ cWorld::cWorld(const AString & a_WorldName, const AString & a_DataPath, eDimensi
 	m_LastChunkCheck(0),
 	m_LastSave(0),
 	m_SkyDarkness(0),
-	m_GameMode(gmNotSet),
+	m_GameMode(gmSurvival),
```
Comment (madmaxoft) [inline src/World.cpp]: This is an unrelated change, should not be in this PR

Comment (peterbell10) [reply src/World.cpp]: It was set in start before, now it is set in the constructor.  This is entirely related.

Comment (madmaxoft) [reply src/World.cpp]: Okay, didn't see that.

**Comment by madmaxoft** [comment]
Does it make sense to keep this separate in two commit, or should we squash-merge?

**Comment by bearbin** [comment]
My thoughts: would it compile and run in the intermediate stage, and do the commits solve logically separate issues? In this case, I think it it could go either way - I would lean towards keeping them separate as they wouldn't be any more confusing that way and more history is always good.

**Comment by madmaxoft** [comment]
I think it doesn't make sense to have this in two separate commits, but I could be wrong.

**Comment by peterbell10** [comment]
The first commit does compile and run.  It fixes the issue that `m_ChunkMap` is `nullptr` but doesn't fix the issue entirely because `SetBlock` will try to interact with the storage thread before it's initialised.
They are very closely related though so having a single commit makes sense.

**Pull request merged**